### PR TITLE
Add sentence-transformers as a named flavor

### PR DESCRIPTION
**Pull request by BenWilson2**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Add basic serialization (save_model, load_model, log_model) and signature default assignment for the sentence-transformers package. 

## How is this patch tested?

- [ ] Existing unit/integration tests
- [X] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

Introduce a sentence-transformers flavor.

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [X] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/sentence_transformers.py [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+    inference_config: Optional[Dict[str, Any]] = None,
+    code_paths: Optional[List[str]] = None,
+    mlflow_model: Optional[Model] = None,
+    signature: Optional[ModelSignature] = None,
+    input_example: Optional[ModelInputExample] = None,
+    pip_requirements: Optional[Union[List[str], str]] = None,
+    extra_pip_requirements: Optional[Union[List[str], str]] = None,
+    conda_env=None,
+    metadata: Dict[str, Any] = None,
+    **kwargs,
```
Comment (WeichenXu123) [inline mlflow/sentence_transformers.py:88]: The **kwargs is not used, shall we remove this argument ?

File: mlflow/sentence_transformers.py [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+) -> None:
+    """
+    Save a trained ``sentence-transformers`` model to a path on the local file system.
+
+    :param model: A trained ``sentence-transformers`` model.
+    :param path: Local path destination for the serialized model to be saved.
+    :param inference_config:
+        A dict of valid overrides that can be applied to a ``sentence-transformer`` model instance
```
Comment (WeichenXu123) [inline mlflow/sentence_transformers.py:96]:
```suggestion
        A dict of valid inference configs that can be applied to a ``sentence-transformer`` model instance and override default inference configs
```

File: mlflow/sentence_transformers.py [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+    else:
+        mlflow_model.signature = _get_default_signature()
+    if input_example is not None:
+        _save_example(mlflow_model, input_example, str(path))
+    if metadata is not None:
+        mlflow_model.metadata = metadata
+
+    model.save(str(model_data_path))
+
+    pyfunc.add_to_model(
```
[the remark this answers is not in the document]
Comment (BenWilson2) [reply mlflow/sentence_transformers.py:151]: We'll have a follow-on PR for pyfunc implementation

**Review by WeichenXu123 — APPROVED** [review]
LGTM

**Pull request merged**